playwright/_impl/_locator.py

```diff
@@ -0,0 +1,454 @@
+        clickCount: int = None,
+        timeout: float = None,
+        force: bool = None,
+        noWaitAfter: bool = None,
+        trial: bool = None,
+    ) -> None:
+        params = locals_to_params(locals())
+        return await self._frame.click(self._selector, strict=True, **params)
+
+    async def dblclick(
+        self,
+        modifiers: List[KeyboardModifier] = None,
+        position: Position = None,
+        delay: float = None,
+        button: MouseButton = None,
+        timeout: float = None,
+        force: bool = None,
+        noWaitAfter: bool = None,
+        trial: bool = None,
+    ) -> None:
+        params = locals_to_params(locals())
+        return await self._frame.dblclick(self._selector, strict=True, **params)
+
+    async def dispatch_event(
+        self,
+        type: str,
+        eventInit: Dict = None,
+        timeout: float = None,
+    ) -> None:
+        params = locals_to_params(locals())
+        return await self._frame.dispatch_event(self._selector, strict=True, **params)
+
+    async def evaluate(
+        self, expression: str, arg: Serializable = None, timeout: float = None
+    ) -> Any:
+        params = locals_to_params(locals())
+        return await self._with_element(
+            lambda h, o: h.evaluate(expression, arg),
+            strict=True,
+            **params,
+        )
+
+    async def evaluate_all(self, expression: str, arg: Serializable = None) -> None:
+        params = locals_to_params(locals())
+        return await self._frame.eval_on_selector_all(self._selector, **params)
+
+    async def evaluate_handle(
+        self, expression: str, arg: Serializable = None, timeout: float = None
+    ) -> "JSHandle":
+        params = locals_to_params(locals())
+        return await self._with_element(
+            lambda h, o: h.evaluate_handle(expression, arg),
+            strict=True,
+            **params,
+        )
+
+    async def fill(
+        self,
+        value: str,
+        timeout: float = None,
+        noWaitAfter: bool = None,
+        force: bool = None,
+    ) -> None:
+        params = locals_to_params(locals())
+        return await self._frame.fill(self._selector, strict=True, **params)
+
+    def locator(
+        self,
+        selector: str,
+    ) -> "Locator":
+        return Locator(self._frame, f"{self._selector} >> {selector}")
+
+    async def element_handle(
+        self,
+        timeout: float = None,
+    ) -> ElementHandle:
+        params = locals_to_params(locals())
+        return cast(
+            ElementHandle,
+            await self._frame.wait_for_selector(
+                self._selector, strict=True, state="attached", **params
+            ),
+        )
+
+    async def element_handles(self) -> List[ElementHandle]:
+        return await self._frame.query_selector_all(self._selector)
+
+    @property
+    def first(self) -> "Locator":
+        return Locator(self._frame, f"{self._selector} >>  _nth=first")
+
+    @property
```

Comment:
Why is this a property? Feels strange that first/last are different from nth().

Reply:
We currently have the following logic which we use everywhere in Python to decide if its a method or property.
```py
        doc_is_property = (
            not method.get("async") and not len(method["args"]) and "type" in method
        )
        if method["name"].startswith("is_") or method["name"].startswith("as_"):
            doc_is_property = False
        if doc_is_property != is_property:
            self.errors.add(f"Method vs property mismatch: {fqname}")
            return
```
-> it needs to be sync, has no arguments, not start with `is` or `as`. Then it's a property. I would keep it instead of adding more edge-cases.